### PR TITLE
per: support multi-char fixed-width strings

### PR DESCRIPTION
**Pull request by xuhe5**
* Include BMPString in the family of KnownMultiplierStringType classes.
* Correct parsing of size-constrained BMPString fields.

### CODE REVIEW

**Comment by xuhe5** [comment]
I have added the required tests. Let me know what you think.

File: asn1tools/codecs/per.py [hunk 1]
```diff
@@ -37,6 +38,31 @@ def is_unbound(minimum, maximum):
...
+    byte_array = bytearray()
+
+    while number_of_bits > 0:
+        byte_array.insert(0, (num & 0xff))
+        num >>= 8
+        number_of_bits -= 8
+
+    return byte_array
+
+
```
Comment (eerimoq) [inline asn1tools/codecs/per.py]: I'm a bit worries about the performance. Encoding and decoding big BMPString (and other string in the same class) will be very slow. Not sure what to do about it.

Comment (xuhe5) [reply asn1tools/codecs/per.py]: I have done some benchmarking. The current code is approximately 50% slower than the previous code on very large fixed-width strings. Decoding of ~30MB of such strings takes about 15-20 seconds on my machine.
It is possible to trade-off performance for complexity, if this is a concern, and speed up the operation for ascii-encoded strings.

Comment (eerimoq) [reply asn1tools/codecs/per.py]: Ok, that was not as bad as I expected. I think that's acceptable for now. I'm sure the performance can be improved later without compromising code complexity too much.

**Pull request merged**